core/vm/interface.go

```diff
@@ -92,3 +93,10 @@ type CallContext interface {
 	// Create creates a new contract
 	Create(env *EVM, me ContractRef, data []byte, gas, value *big.Int) ([]byte, common.Address, error)
 }
+
+// PrecompileManager registers and runs stateful precompiles
+type PrecompileManager interface {
+	IsPrecompile(addr common.Address) bool
+	Run(addr common.Address, input []byte, caller common.Address, value *big.Int, suppliedGas uint64, readonly bool) (ret []byte, remainingGas uint64, err error)
+	Register(addr common.Address, p precompile.StatefulPrecompiledContract) error
+}
```

Comment:
why do we need the interface? are there multiple implementations of this?

Reply:
There are not. I don't recall why I made the interface. Will remove.